### PR TITLE
feat: add tabnew command for opening via a new tab.

### PR DESCRIPTION
**Pull request by tylersaunders**
I'd like to add support for opening a file from the filesystem source in a new tab.

### CODE REVIEW

File: lua/neo-tree/sources/filesystem/commands.lua [hunk 1]
```diff
@@ -94,6 +94,9 @@ end
 M.open_vsplit = function(state)
   cc.open_vsplit(state, utils.wrap(fs.toggle_directory, state))
 end
+M.open_tabnew = function (state)
```
Comment (cseickel) [inline lua/neo-tree/sources/filesystem/commands.lua:97]: This command should also be added to the `buffers` and `git_status` sources to be consistent with the other open commands.

File: README.md [hunk 2]
```diff
@@ -128,6 +128,7 @@ use {
             ["<cr>"] = "open",
             ["S"] = "open_split",
             ["s"] = "open_vsplit",
+            ["t"] = "open_tabnew",
```
Comment (cseickel) [inline README.md:131]: I try to keep the README mappings as a reflection of the default mappings. to that end, can you add this to the `defaults.lua` file as well?

It should also be listed in the help file in `docs/`
See `:h neo-tree-mappings`

**Review by cseickel — COMMENTED** [review]
Thanks for adding, I think it's a good idea to add this. Can yoiu just add the things mentioned in my review comments?

File: lua/neo-tree/sources/filesystem/commands.lua (hunk 1 above)
Comment (tylersaunders) [reply lua/neo-tree/sources/filesystem/commands.lua:97]: Done

File: README.md (hunk 2 above)
Comment (tylersaunders) [reply README.md:131]: Done

**Review by cseickel — APPROVED** [review]
Looks good, thanks!

**Pull request merged**